server-spi/src/main/java/org/keycloak/models/RealmModel.java

```diff
@@ -667,6 +667,15 @@ default GroupModel createGroup(String name, GroupModel toParent) {
      */
     Stream<GroupModel> getTopLevelGroupsStream(Integer first, Integer max);
 
+    /**
+     * Returns top level groups as a stream.
+     * @param first {@code Integer} Index of the first desired group. Ignored if negative or {@code null}.
+     * @param max {@code Integer} Maximum number of returned groups. Ignored if negative or {@code null}.
+     * @param search {@code String} The name of the group that should be search
+     * @return Stream of {@link GroupModel}. Never returns {@code null}.
+     */
+    Stream<GroupModel> getTopLevelGroupsStream(String search, Boolean exact, Integer first, Integer max);
```

Comment:
This method should not be necessary in RealmModel. We should prefer calling GroupProvider directly.